Core/src/Seeding2/DoubletSeedFinder.cpp

```diff
@@ -131,8 +131,14 @@ class Impl final : public DoubletSeedFinder {
       float deltaZ = 0;
       if constexpr (isBottomCandidate) {
         deltaZ = zM - zO;
+        if (outsideRangeCheck( std::abs(deltaZ), 0, middleSpInfo.maxDeltaZbottom)) {
```

Comment:
I think one cut should be enough for both bottom and top and set accordingly in the triplet seeder